doc/policy/packages.md

```diff
@@ -117,3 +118,30 @@ backward-compatible for users and applications that rely on p2p transaction rela
 package validation should not prevent the acceptance of a transaction that would otherwise be
 policy-valid on its own. By always accepting a transaction that passes individual validation before
 trying package validation, we prevent any unintentional restriction of policy.
+
+### Package Replace by Fee
+
+Child-with-unconfirmed-parents packages may replace mempool transactions under the following conditions:
+
+1. The child transaction does not conflict with any transactions in the mempool.
+
+2. The directly conflicting transactions all signal replaceability explicitly.
+
+3. All transactions in the package, including those that do not directly conflict with any mempool
+   transactions, only include an unconfirmed input if that input was included in one of the directly
+   conflicting transactions or is from another transaction in the package.
+
+4. The package fee after deduplication pays an absolute fee of at least the sum paid by the
+   original transactions.
+
+5. The additional fees (difference between absolute fee paid by the package after deduplication and the
+   sum paid by the original transactions) pays for the package's bandwidth after deduplication at or
+   above the rate set by the node's incremental relay feerate. For example, if the incremental relay
+   feerate is 1 satoshi/vB and the package after deduplication contains 500 virtual bytes total, then the
+   package fees after deduplication is at least 500 satoshis higher than the sum of the original transactions.
+
+6. The number of original transactions does not exceed 100.
```

Comment:
I think it could be a limitation for second-layer package issuers, where a child-with-parents package aims to replace more than 5 LN commitment transactions at the same time. A counterparty could extend the 5 non-related commitment transactions descendants until reaching max in-mempool limits, thus interfering with their potential package-RBF.

Am I correct ? That said, I remember we've already said that child-with-parents to replace multiple LN commitment transactions open the door to malicious interferences.

Reply:
I believe this issue can unfortunately arise even when you're replacing a single LN commitment transaction.
Imagine that you're trying to replace `{commitTxAlice, anchorTxAlice}` but Alice has created `anchorTxAlice` to have 100 outputs and broadcast one child tx per output, you're already hitting the 100 transactions limit...

I believe that the only thing that can save Bob here is to broadcast `{commitTxAlice, anchorTxBob}` by relying on the CPFP carve-out rule to get Alice's commitment to confirm. But in order to do that, Bob needs to somehow get the signed `commitTxAlice` (which never appeared in his mempool since his mempool contains `commitTxBob`).

So this rule is an issue for L2 protocols, but it is also a protection against high mempool churn, so it's hard to draw the line.

Reply:
> I think it could be a limitation for second-layer package issuers, where a child-with-parents package aims to replace more than 5 LN commitment transactions at the same time. A counterparty could extend the 5 non-related commitment transactions descendants until reaching max in-mempool limits, thus interfering with their potential package-RBF.

Yeah, batching multiple commitment transaction replacements isn't very safe. That's why we have the "

Warning: Batched fee-bumping may be unsafe for some use cases. Users and application developers should take caution if utilizing multi-parent packages." in the [doc](https://github.com/bitcoin/bitcoin/blob/master/doc/policy/packages.md#package-mempool-acceptance-rules).

> I believe this issue can unfortunately arise even when you're replacing a single LN commitment transaction.

I don't think this scenario is possible with our current mempool policy. With our descendant limit at 25,  `anchorTxAlice` wouldn't be able to have 100 children.